### PR TITLE
Addition of basic Auth functionality as mentioned in issue Basic Auth #330 

### PR DESCRIPTION
**Pull request by hkiiita**
Addition of basic Auth functionality as mentioned in issue Basic Auth #330

### CODE REVIEW

**Comment by hkiiita** [comment]
Incorporates multiple users.

File: internal/creds/processCreds.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	Users []Credential `yaml:"users"`
+}
+
+var CredsFromFile CredentialsList
+
+func GetCredentials() (CredentialsList, error) {
+	if len(CredsFromFile.Users) == 0 {
+		slog.Error("CredentialsList is empty", "error", errors.New("Credentials are empty."))
+		return CredentialsList{}, errors.New("Credentials are empty.")
+	} else {
```
Comment (susarlanikhilesh) [inline internal/creds/processCreds.go:23]: else is not required

**Comment by dfarr** [comment]
Thank you so much for the PR @hkiiita 🎉 

We can't believe it, but we actually got two PRs for the same feature. We are going to merge #332, but would you be up for incorporating multiple users (which is lacking in #332) once that one is merged?

**Comment by hkiiita** [comment]
> Thank you so much for the PR @hkiiita 🎉
> 
> We can't believe it, but we actually got two PRs for the same feature. We are going to merge #332, but would you be up for incorporating multiple users (which is lacking in #332) once that one is merged?

Hi @dfarr thank you so much for your kind reply on this. Yes , i will be more than happy to continue to work on this after 332 gets merged. I have already incorporated multiple users authentication into my PR #333 , so will wait on your side for a review and further steps. Thanks again !

**Comment by dfarr** [comment]
Hey @hkiiita, PR #332 has been merged, if you're still up for it, we could extend basic auth to support multiple users - I believe we would want to extend [these arguments](https://github.com/resonatehq/resonate/blob/main/cmd/serve/serve.go#L193-L194) to take an array of username/password combos (and propagate that array down accordingly)

**Comment by hkiiita** [comment]
Hi @dfarr , thanks for the reply . Yes I definitely would want to work on this . Just wanted to share that, if you are ok with it ,  I will reach out to you on 18th June.

**Comment by hkiiita** [comment]
https://github.com/resonatehq/resonate/pull/352

**Pull request closed without merging**